### PR TITLE
Sort the list of Request.meta alphabetically #5061

### LINKED ISSUE #5061: Sort the list of Request.meta keys alphabetically
[The list](https://docs.scrapy.org/en/latest/topics/request-response.html#request-meta-special-keys) is rather long, and getting longer with time, so I think it would be best to keep it sorted alphabetically, to make lookup easier.

### PR DESCRIPTION
**Pull request by pratik1500**
Creating the pull request for Sort the list of Request.meta keys alphabetically #5061 sorted the file request-response.rst and arranged them in order.

Resolves #5061

### CODE REVIEW

File: docs/topics/request-response.rst [hunk 1]
```diff
@@ -363,26 +363,28 @@ are some special keys recognized by Scrapy and its built-in extensions.
 
 Those are:
 
-* :reqmeta:`dont_redirect`
-* :reqmeta:`dont_retry`
-* :reqmeta:`handle_httpstatus_list`
-* :reqmeta:`handle_httpstatus_all`
-* :reqmeta:`dont_merge_cookies`
+
```
Comment (Gallaecio) [inline docs/topics/request-response.rst:366]: We should avoid this empty line.

File: docs/topics/request-response.rst [hunk 2]
```diff
@@ -363,26 +363,28 @@ are some special keys recognized by Scrapy and its built-in extensions.
...
+* ``ftp_password`` (See :setting:`FTP_PASSWORD` for more info)
```
Comment (Gallaecio) [inline docs/topics/request-response.rst:367]: These `ftp_…` entries should not go before `bindaddress`.

**Review by Gallaecio — COMMENTED** [review]
Could you also [address this](https://github.com/scrapy/scrapy/issues/5061#issuecomment-804008762)? We need to find where in the documentation users would expect information on how to handle cookies, and mention those two request meta keys to make it easier for users to find them (together) in the documentation.

File: docs/topics/request-response.rst (hunk 2 above)
Comment (pratik1500) [reply docs/topics/request-response.rst]: Have changed the location of `ftp`

Comment (Gallaecio) [reply docs/topics/request-response.rst]: Their new location is not right either. They should be between `download_timeout` and `handle_httpstatus_all`.

Comment (pratik1500) [reply docs/topics/request-response.rst]: Have made changes, please look at it.

Comment (pratik1500) [reply docs/topics/request-response.rst]: Have made changes, please look at it

File: docs/topics/request-response.rst [hunk 3]
```diff
@@ -363,26 +363,28 @@ are some special keys recognized by Scrapy and its built-in extensions.
...
+* ``ftp_user`` (See :setting:`FTP_USER` for more info)
+* :reqmeta:`handle_httpstatus_all`
+* :reqmeta:`handle_httpstatus_list`
 * :reqmeta:`max_retry_times`
+* :reqmeta:`proxy`
+* :reqmeta:`redirect_reasons`
+* :reqmeta:`redirect_urls`
+* :reqmeta:`referrer_policy`
+
+
```
Comment (Gallaecio) [inline docs/topics/request-response.rst:388]: Could you leave a single empty line here? Those 2 new empty lines do not seem necessary.

**Comment by Gallaecio** [comment]
Please, do not miss https://github.com/scrapy/scrapy/pull/5065#pullrequestreview-618600713

**Comment by Gallaecio** [comment]
Please, see https://github.com/scrapy/scrapy/pull/5065#pullrequestreview-618600713

**Comment by pratik1500** [comment]
Sorry for this I am not able to get the  https://github.com/scrapy/scrapy/issues/5061#issuecomment-804008762, please would you help me in elaborating it.

**Review by Gallaecio — APPROVED** [review]
I’ve just reviewed existing references, and both meta keys seem discoverable from [what I would consider the main entry point for cookies in the documentation](https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#cookies-mw), so as far as I’m concern there are no additional changes needed.

@kmike What do you think? Should we make sure we mention them somewhere else?

**Comment by pratik1500** [comment]
Thanks a lot

**Comment by kmike** [comment]
Thanks for the fix @pratik1500, and thanks for checking it @Gallaecio!

**Pull request merged**